packages/firestore/src/remote/datastore.ts

```diff
@@ -51,36 +43,11 @@ interface CommitRequest extends api.CommitRequest {
  */
 export class Datastore {
   constructor(
-    private queue: AsyncQueue,
     private connection: Connection,
     private credentials: CredentialsProvider,
     private serializer: JsonProtoSerializer
   ) {}
 
-  newPersistentWriteStream(
```

Comment:
Rather than changing the object models and completely rearranging the client (so that it doesn't match the other ports), a better strategy would be to adopt tree-shakeable APIs internally.

That is, make classes like `Datastore` more like just dumb structs, and make the methods that used to be in the class plain functions that take `Datastore` as an argument.

This will greatly reduce the amount of churn in the relative arrangement of components. Indeed, you could apply this to `commit` and other methods on datastore to ensure that all of it tree-shakes out if unused.

Reply:
One of the downsides is that it makes every member public, and I kind of wanted to avoid this. Your argument that we should pull out all methods does win here though. I updated the PR, which also greatly reduces the diff (a large part are now just test cleanup).

Reply:
I pushed a commit with a TypeScript version of this hack: https://github.com/firebase/firebase-js-sdk/pull/2971/commits/0295cd5df3bd7ee7f30f8ecb18b174e1f35e8308

It should be pretty easy to understand, but I wanted to point out a downside: With this change, at least IntelliJ now consider all members of `DatastoreImpl` unused, which would breaks its automated refactor should we use it on classes like this.

Reply:
That's a lot better than my suggestion. If we can get IntelliJ happy, I think this is the way to go.